tiledb/query_condition.py

```diff
@@ -165,21 +179,38 @@ def visit_Compare(self, node):
                     f"Type mismatch between attribute `{att}` and value `{val}`."
                 )
 
-        c_obj = qc.qc(self._ctx)
+        result = qc.qc(self._ctx)
 
-        if not hasattr(c_obj, f"init_{dtype_name}"):
+        if not hasattr(result, f"init_{dtype_name}"):
             raise tiledb.TileDBError(
                 f"PyQueryCondition's `init_{dtype_name}` not found."
             )
 
-        init_qc = getattr(c_obj, f"init_{dtype_name}")
+        init_qc = getattr(result, f"init_{dtype_name}")
 
         try:
             init_qc(att, val, op)
         except tiledb.TileDBError as e:
             raise tiledb.TileDBError(e)
 
-        return c_obj
+        return result
+
+    def visit_BinOp(self, node):
+        AST_TO_TILEDB = {ast.BitAnd: qc.TILEDB_AND}
+
+        try:
+            op = AST_TO_TILEDB[type(node.op)]
+        except KeyError:
+            raise tiledb.TileDBError(
+                "Unsupported binary operator. Only & is currently supported."
```

Comment:
Could we print the op here to make the error message a little bit more specific?